src/utils/getQueryStringFromUrl.ts

```diff
@@ -0,0 +1,9 @@
+import { NextRouter } from 'next/router';
+
+export const getQueryStringFromUrl = (router: NextRouter) => {
```

Comment:
This function don't need to know about next router. You can pass just a string (q part) and get necessary details.